### ISSUE TITLE
support registration of eth withdrawal address using a transaction

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain#217]
Part of story <https://github.com/threefoldtech/home/issues/56>.

Will use Rivine encoding.

Required to unblock Mobicage team

**Comment by GlenDC** [comment]
Proposed Tx:

```javascript
// EthereumAddressRegistrationTransaction defines the Transaction (with version 0xD0)
// used to register an ethereum address in tfchain AND ethereum,
// as to couple a tfchain address with an ethereum address
EthereumAddressRegistrationTransaction struct {
	// Register a TFT address,
	// which will be coupled with the Ethereum Generated
	// using the last 20 bytes of the blake2b hash of the given TFT address
	Address types.UnlockHash

	// TransactionFee defines the regular Tx fee.
	TransactionFee types.Currency `json:"txfee"`

	// OracleFee defines the oracle fee.
	OracleFee types.Currency `json:"oraclefee"`

	// CoinInputs are only used for the required fees,
	// which contains the regular Tx fee as well as the additional fees,
	// to be paid for the address registration. At least one CoinInput is required.
	CoinInputs []types.CoinInput `json:"coininputs"`
	// RefundCoinOutput is an optional coin output that can be used
	// to refund coins paid as inputs for the required fees.
	RefundCoinOutput *types.CoinOutput `json:"refundcoinoutput,omitempty"`
}
```

**Comment by LeeSmet** [comment]
I'm not really sure if we want to allow anyone to register any address. Perhaps it is better to use the unlockhash of the first coin input?

**Comment by GlenDC** [comment]
> I'm not really sure if we want to allow anyone to register any address. Perhaps it is better to use the unlockhash of the first coin input?

Seems a bit too implicit. I think I'll simply go for the registration of the public key with a signature. From that public key it is easy enough to generate the unlock hash.

**Comment by robvanmieghem** [comment]
The reason for the registration of withdrawal adresses is than an ethereum adress is just 20 bytes, the contract would not be able to see the difference between a transfer and a withdrawal if withdrawing from an exchange without this.
Also, there is no free choice of ethereum address since it would be possible to steal someone else's token otherwise and requiring an ethereum private key and signature would only complicate things a lot.

**Comment by GlenDC** [comment]
Closed, follow-up issue has been created regarding the Registration Fee: <https://github.com/threefoldfoundation/tfchain/issues/237>
